modules/web/src/de/diedavids/cuba/ceuesr/web/EntitySoftReferenceLookupAction.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2008-2018 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.diedavids.cuba.ceuesr.web;
+
+import com.haulmont.chile.core.model.MetaClass;
+import com.haulmont.cuba.core.entity.Entity;
+import com.haulmont.cuba.core.global.Metadata;
+import com.haulmont.cuba.gui.Dialogs;
+import com.haulmont.cuba.gui.UiComponents;
+import com.haulmont.cuba.gui.actions.picker.LookupAction;
+import com.haulmont.cuba.gui.app.core.inputdialog.InputDialog;
+import com.haulmont.cuba.gui.app.core.inputdialog.InputParameter;
+import com.haulmont.cuba.gui.components.ActionType;
+import com.haulmont.cuba.gui.components.Component;
+import com.haulmont.cuba.gui.components.LookupField;
+import com.haulmont.cuba.gui.screen.*;
+import de.diedavids.cuba.ceuesr.entity.Commentable;
+import de.diedavids.cuba.ceuesr.web.comment.MetadataSelector;
+
+import javax.inject.Inject;
+import java.util.Map;
+
+@ActionType(EntitySoftReferenceLookupAction.ID)
+public class EntitySoftReferenceLookupAction extends LookupAction {
```

Comment:
Custom lookup action so that before calling the lookup screen, a selection can be made which entity should be looked up.